### PR TITLE
[OneDNN][PIR] Add conv2d_transpose_bn_fuse_pass

### PR DESCRIPTION
**Pull request by zhanglirong1999**
### PR Category
Others

### PR Types
New features

### Description
Add conv2d_transpose_bn_fuse_pass and conv2d_transpose_eltwiseadd_bn_fuse_pass in OneDNN. Implemented the previous cpu pass conv2d_transpose_bn_fuse_pass and conv2d_transpose_eltwiseadd_bn_fuse_pass

### CODE REVIEW

**Review by LLee233 — DISMISSED** [review]
Should be more appropriate if changing the name to "conv2d_transpose_bias_bn_fuse_pass" since conv2d_transpose only supports bias_add

**Comment by zhanglirong1999** [comment]
@LLee233 ,yes, I think so. But the CI is pass, I think it can modify if there are other problems. If there are no other changes, I will modify them in the next pass PR.

File: paddle/fluid/pir/transforms/onednn/conv2d_transpose_bn_fuse_pass.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/onednn/conv2d_transpose_bn_fuse_pass.h]: 2023 -> 2024

File: test/ir/pir/fused_pass/onednn/test_conv_transpose_bn_onednn_fuse_pass.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (yuanlehome) [inline test/ir/pir/fused_pass/onednn/test_conv_transpose_bn_onednn_fuse_pass.py]: 2023 -> 2024

**Review by yuanlehome — APPROVED** [review]

**Pull request merged**